apps/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -1660,6 +1738,59 @@ export class RushConfiguration {
     return undefined;
   }
 
+  private _collectVersionsForDependencies(
+    versionsForDependencies: Map<string, Set<string>>,
+    dependencies: ReadonlyArray<PackageJsonDependency>,
+    cyclicDependencies: Set<string>,
+    variant: string | undefined
+  ): void {
+    const commonVersions: CommonVersionsConfiguration = this.getCommonVersions(variant);
+    const allowedAlternativeVersions: Map<string, ReadonlyArray<string>> =
+      commonVersions.allowedAlternativeVersions;
+
+    for (const dependency of dependencies) {
+      const alternativesForThisDependency: ReadonlyArray<string> =
+        allowedAlternativeVersions.get(dependency.name) || [];
+
+      // For each dependency, collectImplicitlyPreferredVersions() is collecting the set of all version specifiers
+      // that appear across the repo.  If there is only one version specifier, then that's the "preferred" one.
+      // However, there are a few cases where additional version specifiers can be safely ignored.
+      let ignoreVersion: boolean = false;
+
+      // 1. If the version specifier was listed in "allowedAlternativeVersions", then it's never a candidate.
+      //    (Even if it's the only version specifier anywhere in the repo, we still ignore it, because
+      //    otherwise the rule would be difficult to explain.)
+      if (alternativesForThisDependency.indexOf(dependency.version) > 0) {
+        ignoreVersion = true;
+      } else {
+        // Is it a local project?
+        const localProject: RushConfigurationProject | undefined = this.getProjectByName(dependency.name);
+        if (localProject) {
+          // 2. If it's a symlinked local project, then it's not a candidate, because the package manager will
+          //    never even see it.
+          // However there are two ways that a local project can NOT be symlinked:
+          // - if the local project doesn't satisfy the referenced semver specifier; OR
+          // - if the local project was specified in "cyclicDependencyProjects" in rush.json
+          if (
+            semver.satisfies(localProject.packageJsonEditor.version, dependency.version) &&
+            !cyclicDependencies.has(dependency.name)
```

Comment:
```suggestion
            !cyclicDependencies.has(dependency.name) &&
            semver.satisfies(localProject.packageJsonEditor.version, dependency.version)
```

I'm sure `cyclicDependencies.has` is a cheaper call than `semver.satisfies`.